pkg/daemon/not_cos.go

```diff
@@ -0,0 +1,60 @@
+package daemon
+
+import (
+	"io/ioutil"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// NotCoreOSClient is a wrapper around RpmOstreeClient that implements NodeUpdaterClient
+// safely on unsupported and legacy Operating systems.
+type notCoreOSClient struct {
+	r *RpmOstreeClient
+}
+
+// NewNodeUpdaterClientNotCoreOS returns a NodeUpdaterClient for legacy and non-CoreOS
+// operating systems like RHEL 7.
+func NewNodeUpdaterClientNotCoreOS() NodeUpdaterClient {
+	return &notCoreOSClient{
+		r: &RpmOstreeClient{},
+	}
+}
+
+var (
+	// notCoreOSClient is a NodeUpdaterClient.
+	_ NodeUpdaterClient = &notCoreOSClient{}
+
+	// ErrNotCoreosVariant is a generic error for un-supported tasks on legacy systems.
+	ErrNotCoreosVariant = errors.New("operating system is not a CoreOS Variant")
+)
+
+// GetBootedDeployment returns the booted demployment.
+func (noCos *notCoreOSClient) GetBootedDeployment() (*RpmOstreeDeployment, error) {
```

Comment:
Note: none of this is "safe" yet..I imagine that we should just return errors for all this. To get started, I thought the least disruptive/buggy approach is just to pass through a `RpmOstreeClient` for now.